src/main/java/org/opentripplanner/standalone/config/buildconfig/OsmConfig.java

```diff
@@ -40,42 +40,56 @@ local filesystem (including cloud storage services) or at various different loca
           the local filesystem.
           """
         )
-        .asObjects(OsmConfig::mapOsmExtractConfig)
+        .asObjects(nodeAdapter -> mapOsmParameters(nodeAdapter, defaults))
     );
   }
 
-  private static OsmExtractParameters mapOsmExtractConfig(NodeAdapter config) {
-    var builder = new OsmExtractParametersBuilder();
-    builder.withSource(mapSource(config));
-    builder.withTimeZone(mapTimeZone(config));
-    builder.withOsmTagMapper(mapTagMapping(config));
-    return builder.build();
-  }
-
-  private static URI mapSource(NodeAdapter config) {
-    return config
-      .of("source")
-      .since(V2_2)
-      .summary("The unique URI pointing to the data file.")
-      .asUri();
-  }
-
-  private static ZoneId mapTimeZone(NodeAdapter config) {
-    return config
-      .of("timeZone")
-      .since(V2_2)
-      .summary(
-        "The timezone used to resolve opening hours in OSM data. Overrides the value specified in osmDefaults."
+  public static OsmExtractParameters mapOsmParameters(
+    NodeAdapter node,
+    OsmExtractParameters defaults
+  ) {
+    return defaults
+      .copyOf()
+      .withSource(
+        node.of("source").since(V2_2).summary("The unique URI pointing to the data file.").asUri()
       )
-      .asZoneId(null);
+      .withOsmTagMapper(
+        node
+          .of("osmTagMapping")
+          .since(V2_2)
+          .summary(
+            "The named set of mapping rules applied when parsing OSM tags. Overrides the value specified in `osmDefaults`."
+          )
+          .asEnum(defaults.osmTagMapper())
+      )
+      .withTimeZone(
+        node
+          .of("timeZone")
+          .since(V2_2)
+          .summary(
+            "The timezone used to resolve opening hours in OSM data. Overrides the value specified in `osmDefaults`."
+          )
+          .asZoneId(defaults.timeZone().orElse(null))
+      )
+      .build();
   }
 
-  private static OsmTagMapper mapTagMapping(NodeAdapter node) {
-    return node
-      .of("osmTagMapping")
-      .since(V2_2)
-      .summary("The named set of mapping rules applied when parsing OSM tags.")
-      .asEnum(OsmTagMapper.Source.DEFAULT)
-      .getInstance();
+  public static OsmExtractParameters mapOsmGenericParameters(NodeAdapter node) {
```

Comment:
Should we use this method from `mapOsmParameters`, so that when adding a new parameter it should be added only once. We could have a default instance, which we could pass as defaults, not having to define the defaults defined here, but in a constant, similar to `NetexFeedParameters.DEFAULT`

Reply:
I looked at the netex implementation but I thought it complicates things a bit by adding a new layer of defaults even though they are not really ever used. I was more thinking that should we document for all parameters that can be overridden by defaults that the defaults exist like it's done in this pr. Although, using the netex defaults way of doing things doesn't stop us from appending certain parameters with the extra documentation.

Reply:
Re-read your comment, unlike what I previously commented, the defaults are actually used. I think I will refactor this a bit.

Reply:
Is this ok https://github.com/opentripplanner/OpenTripPlanner/pull/4635/commits/f4f99d2c790c55d42b47d363df6b4deb70d38f0c ?

Reply:
Exactly